### ISSUE TITLE
Privoxy fails to start due to duplicate listen address in config file

### ISSUE THREAD

**Issue by Mossop** [haugene/docker-transmission-openvpn#2692]
### Is there a pinned issue for this?

- [X] I have read the pinned issues and could not find my issue

### Is there an existing or similar issue/discussion for this?

- [X] I have searched the existing issues
- [X] I have searched the existing discussions

### Is there any comment in the documentation for this?

- [X] I have read the documentation, especially the FAQ and Troubleshooting parts

### Is this related to a provider?

- [X] I have checked the provider repo for issues
- [X] My issue is NOT related to a provider

### Are you using the latest release?

- [X] I am using the latest release

### Have you tried using the dev branch latest?

- [X] I have tried using dev branch

### Docker run config used

`docker-compose.yml`:

```
  transmission:
    container_name: transmission
    image: haugene/transmission-openvpn:edge
    restart: always

    dns:
      - 1.1.1.1
      - 1.0.0.1

    networks:
      - services

    cap_add:
      - NET_ADMIN

    devices:
      - /dev/net/tun

    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /var/lib/transmission/:/config/transmission-home
      - /pool/media/:/pool/media

    environment:
      OPENVPN_PROVIDER: "PIA"
      OPENVPN_CONFIG: "uk_london"
      PIA_OPENVPN_CONFIG_BUNDLE: "openvpn-tcp"
      OPENVPN_USERNAME: "****"
      OPENVPN_PASSWORD: "****"
      ENABLE_UFW: "true"
      LOCAL_NETWORK: "10.10.0.0/16,172.16.0.0/12"
      PUID: "137"
      PGID: "151"
      TRANSMISSION_DOWNLOAD_DIR: "/pool/media/torrents/completed"
      TRANSMISSION_INCOMPLETE_DIR: "/pool/media/torrents/incomplete"
      TRANSMISSION_WATCH_DIR: "/pool/media/torrents/watch"
      TRANSMISSION_WEB_UI: flood-for-transmission
      WEBPROXY_ENABLED: "true"
      WEBPROXY_PORT: "8888"
```

### Current Behavior

Privoxy fails to start, logs the following error:
```
2023-08-30 11:15:52.884 7f5f921d0b80 Fatal error: can't bind to 10.10.12.131:8888: There may be another Privoxy or some other proxy running on port 8888
```

### Expected Behavior

Privoxy should start as normal.

### How have you tried to solve the problem?

Tried the edge tag. Looked at the privoxy config file and spotted the issue, see discussion below.

### Log output

```
2023-08-30 11:15:52.884 7f5f921d0b80 Fatal error: can't bind to 10.10.12.131:8888: There may be another Privoxy or some other proxy running on port 8888
```

### HW/SW Environment

```markdown
- OS: Linux
- Docker: version 24.0.5, build ced0996
```

### Anything else?

Appears to have been introduced by this change to the privoxy startup script in #2610.

The config file looks like this before the script runs:

```
listen-address  127.0.0.1:8118
listen-address  [::1]:8118
```

Before the change above the first sed script would replace the IPv4 listen address with the correct one and the second sed script would remove the IPv6 listen address. Now the first sed script replaces both the listen addresses and the second sed script does nothing. This tells privoxy to attempt to listen on the same address twice causing it to fail.

**Comment by resuarez** [comment]
Same problem here

**Comment by ilike2burnthing** [comment]
@heavygee ^

**Comment by RafalHo** [comment]
Same problem here

**Comment by sbe-arg** [comment]
hahaha same

**Comment by pdobrien** [comment]
there is a duplicate listen instruction in the privoxy config file.  Remove or comment out one of them and restart.

**Comment by nleach7** [comment]
> there is a duplicate listen instruction in the privoxy config file. Remove or comment out one of them and restart.

My proxy settings are defined as environment variables. Is a config file something I should migrate to?

**Comment by pdobrien** [comment]
I'm talking about the privoxy config file that's generated on container start (/etc/privoxy/config inside the container).  Mine contained two listen statements like so:

listen-address 172.17.0.6:8888                                                   
listen-address 172.17.0.6:8888   

I commented one of them out and restarted the container, and privoxy was able to start normally.  One of the changes in the latest release was to privoxy/scripts/start.sh which I assume caused this issue.

**Comment by pkishino** [comment]
@heavygee can you ctake a look at this as you authored the PR, thanks

**Comment by edgd1er** [comment]
the new way to set the ip is too generic and change both listen lines (ipv4 and ipv6):
https://github.com/haugene/docker-transmission-openvpn/blob/48299204d160d2d699ce6dd5ccdfcba0ecf13c83/privoxy/scripts/start.sh#L30

Either move https://github.com/haugene/docker-transmission-openvpn/blob/48299204d160d2d699ce6dd5ccdfcba0ecf13c83/privoxy/scripts/start.sh#L32-L33 to line 26, before ipv4 modification, or fix the current sed.

here is an example of the replacement I use on another project using privoxy:
```
# Set the port for the IPv4 interface
  adr=$(ip -4  a show eth0| grep -oP "(?<=inet )([^/]+)")
  adr=${adr:-"0.0.0.0"}
  sed -i -E "s/^listen-address\s+[0-9]{1,3}.*/listen-address ${adr}:${port}/" "$2"
```

I'm sorry, i'm not using anymore this project and my fork is too much different from this one to send a PR.

**Comment by heavygee** [comment]
Yeah, my bad on that too-grabby sed 🙄 - the #2678 PR has the better fixes in it - can that PR get approved and merged?

**Comment by pkishino** [comment]
can all of you please pull :dev branch and test it there, if everything is fine I'll release an update..

**Comment by Mossop** [comment]
> can all of you please pull :dev branch and test it there, if everything is fine I'll release an update..

This seems to work for me.

**Comment by juliemars** [comment]
> can all of you please pull :dev branch and test it there, if everything is fine I'll release an update..

yep working for me as well thanks!

**Comment by oregonpillow** [comment]
@pkishino would you be open to tests in the pipeline to prevent this kind of issue in the future? 

Would you be open to creating CI tests that:
1. downloads a small public torrent
2. check the file hash post download to confirm torrenting/vpn functionality worked
3. a test to confirm privoxy connections work.

i would be happy to help on this.

**Comment by pkishino** [comment]
Hey, sure,
That sounds interesting.

**Comment by usma0118** [comment]
+1 facing same issue.

**Comment by ilike2burnthing** [comment]
@usma0118 https://github.com/haugene/docker-transmission-openvpn/issues/2692#issuecomment-1703430324

**Comment by pkishino** [comment]
fixed by #2696
